### PR TITLE
Deprecate .unsafe functions of typed_actor

### PR DESCRIPTION
**Pull request by ufownl**
@Neverlord it seems that you forgot `typed_actor`. ;)

### CODE REVIEW

File: libcaf_core/caf/function_view.hpp [hunk 1]
```diff
@@ -195,16 +193,21 @@ class function_view {
...
-    if (impl_.unsafe() && !x.unsafe())
+    if (!impl_ && x)
       new_self(x);
-    if (!impl_.unsafe() && x.unsafe())
+    if (impl_ && !x)
       self_.~scoped_actor();
     impl_.swap(x);
   }
 
+  void cleanup() {
```
Comment (Neverlord) [inline libcaf_core/caf/function_view.hpp]: I'm not a big fan of the name `cleanup`. It's not used anywhere else in CAF. Maybe `reset`?

File: libcaf_io/test/remote_spawn.cpp [hunk 2]
```diff
@@ -89,7 +89,7 @@ void run_client(int argc, char** argv, uint16_t port) {
   auto f1 = make_function_view(*calc);
   CAF_REQUIRE_EQUAL(f1(add_atom::value, 10, 20), 30);
   CAF_REQUIRE_EQUAL(f1(sub_atom::value, 10, 20), -10);
-  f1.assign(unsafe_actor_handle_init);
+  f1.cleanup();
```
Comment (Neverlord) [inline libcaf_io/test/remote_spawn.cpp]: Same here, I think `f1.reset()` reads better.

**Comment by Neverlord** [comment]
Obviously I didn't forget it. It was a test to see how long it would take you to make a PR. 😁

Congratulation, btw. You've passed the test. 😉

**Comment by ufownl** [comment]
Maybe I have naming difficulty disease. 😂
Btw, tweak has been done.

**Pull request merged**